examples/platform/silabs/display/lcd.cpp

```diff
@@ -149,11 +147,102 @@ void SilabsLCD::WriteDemoUI()
     }
 }
 
+void SilabsLCD::WriteStatus()
+{
+    uint8_t lineNb = 0;
+    char str[20];
+
+    GLIB_clear(&glibContext);
+    sprintf(str, "# Fabrics : %d", mStatus.nbFabric);
```

Comment:
maybe StringBuilder would be safer than trying to have sizes guaranteed to match.

Then this is super safe and less worry below about various strlen. Just reset the StringBuilder and use its c_str that is guaranteed to be null terminated. You may even add `AddMarkerIfOverflow`

Reply:
Will investigate in a follow up, LCD needs rework anyway